From ff1c8342fe99ea1ae7ddb0accad61292fd7b3143 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <aalonso@freescale.com>
Date: Tue, 17 Apr 2012 14:18:59 -0500
Subject: [PATCH 313/449] ENGR00179880 FSL-UT-ESAI-006: esai play record
 concurrent test case

* Add ESAI FSL-UT-ESAI-006 use test case
  Test concurrent playback/record for alsa driver with
  ESAI interface; Play wav file and send process to
  background, record new audio file.

Signed-off-by: Adrian Alonso <aalonso@freescale.com>
---
 test/doc/ut_um.toc                         |  1 +
 test/mxc_esai_test/doc/FSL-UT-ESAI-006.txt | 40 ++++++++++++++++++++++++++++++
 2 files changed, 41 insertions(+)
 create mode 100644 test/mxc_esai_test/doc/FSL-UT-ESAI-006.txt

diff --git a/test/doc/ut_um.toc b/test/doc/ut_um.toc
index 118acc6..e0ff59b 100644
--- a/test/doc/ut_um.toc
+++ b/test/doc/ut_um.toc
@@ -27,6 +27,7 @@ include::../mxc_esai_test/doc/FSL-UT-ESAI-002.txt[]
 include::../mxc_esai_test/doc/FSL-UT-ESAI-003.txt[]
 include::../mxc_esai_test/doc/FSL-UT-ESAI-004.txt[]
 include::../mxc_esai_test/doc/FSL-UT-ESAI-005.txt[]
+include::../mxc_esai_test/doc/FSL-UT-ESAI-006.txt[]
 
 FEC
 ---
diff --git a/test/mxc_esai_test/doc/FSL-UT-ESAI-006.txt b/test/mxc_esai_test/doc/FSL-UT-ESAI-006.txt
new file mode 100644
index 0000000..1875b56
--- /dev/null
+++ b/test/mxc_esai_test/doc/FSL-UT-ESAI-006.txt
@@ -0,0 +1,40 @@
+FSL-UT-ESAI-006
+~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Test concurrent playback/record for alsa driver with ESAI interface;
+Play wav file and send process to background, record new audio file.
+
+| Automated |
+NO
+
+| Kernel Config Option |
+ CONFIG_IMX_HAVE_PLATFORM_IMX_ESAI=y
+ CONFIG_IMX_HAVE_PLATFORM_IMX_I2C=y
+ CONFIG_SND_SOC_IMX_CS42888=y
+ CONFIG_SND_SOC_CS42888=y
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+SabreAI
+
+| Test Procedure |
+. run the following commands on target board,
+
+ $ aplay -Dhw:0,0 file.wav >& /dev/null \| arecord -Dhw:0,0 -f S16_LE -r 48000 -c 2 rec-S16LE-48k.wav
+ $ aplay -Dhw:0,0 rec-S16LE-48k.wav
+
+| Expected Result |
+The sound is heard clearly while recording an additional audio file.
+The record sound file is heard clearly and properly.
+
+|====================================================================
+
+<<<
+
-- 
1.8.0

